Controllers/ImageController.cs

```diff
@@ -0,0 +1,336 @@
+            {
+                return new ObjectResult(ex.ToString()) { StatusCode = (int?)HttpStatusCode.InternalServerError };
+            }
+        }
+
+        /// <summary>
+        /// Generates a list of images from the specified base image.
+        /// Expected arguments:
+        /// - baseImage: the image as a multipart form POST. This is the image from which all other images will be generated. Should generally be 512x512 or larger, ideally in PNG format.
+        /// - platform: a list of values specifying the platform(s) for which the images are being generated, e.g. "windows10"
+        /// - padding: a value between 0 and 1 specifying the padding for the generated images.
+        /// - color: a hex color value to use as the background color of the generated images. If null, a best guess -- the color of pixel (0,0) -- will be used.
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost("generateImagesZip")]
+        public async Task<ActionResult> Post([FromForm] ImageFormData Form)
+        {
+            var zipId = Guid.NewGuid();
+
+            try
+            {
+                using (var args = ImageGenerationModel.FromFormData(HttpContext.Request.Form, HttpContext.Request.Form.Files))
+                {
+                    // Punt if we have invalid arguments.
+                    if (!string.IsNullOrEmpty(args.ErrorMessage))
+                    {
+                        return new ObjectResult(args.ErrorMessage) { StatusCode = (int?)HttpStatusCode.BadRequest };
+                    }
+
+                    var profiles = GetProfilesFromPlatforms(args.Platforms);
+                    var imageStreams = new List<Stream>(profiles.Count);
+
+                    using (var zip = ZipFile.Open(CreateFilePathFromId(zipId), ZipArchiveMode.Create))
+                    {
+                        var iconObject = new IconRootObject();
+                        foreach (var profile in profiles)
+                        {
+                            var stream = CreateImageStream(args, profile);
+                            imageStreams.Add(stream);
+                            var fmt = string.IsNullOrEmpty(profile.Format) ? "png" : profile.Format;
+                            var iconEntry = zip.CreateEntry(profile.Name + "." + fmt, CompressionLevel.Fastest);
+                            var iconStream = iconEntry.Open();
+                            await stream.CopyToAsync(iconStream);
+                            iconStream.Close();
+                            stream.Close();
+        
+                            iconObject.icons.Add(new IconObject(profile.Folder + profile.Name + "." + fmt, profile.Width + "x" + profile.Height));
+                        }
+
+                        var iconStr = JsonConvert.SerializeObject(iconObject, Formatting.Indented);
+
+                        using (StreamWriter writer = new StreamWriter(zip.CreateEntry("icons.json", CompressionLevel.Optimal).Open()))
```

Comment:
`icons.json` could be a constant I guess too? Not sure what's definining the specific names for these things, just assuming the service contract you're making with PWA Builder?

Reply:
Fixed